File: persistence/src/main/java/io/sparkled/persistence/stage/impl/query/GetAllStagesQuery.java
package io.sparkled.persistence.stage.impl.query;

import io.sparkled.model.entity.Stage;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.List;

public class GetAllStagesQuery implements PersistenceQuery<List<Stage>> {

    @Override
    public List<Stage> perform(QueryFactory queryFactory) {
        return queryFactory
                .selectFrom(qStage)
                .orderBy(qStage.name.asc())
                .fetch();
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/stage/impl/query/GetAllStagesQuery.kt
package io.sparkled.persistence.stage.impl.query

import io.sparkled.model.entity.QStage.Companion.stage
import io.sparkled.model.entity.Stage
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory

class GetAllStagesQuery : PersistenceQuery<List<Stage>> {

    override fun perform(queryFactory: QueryFactory): List<Stage> {
        return queryFactory
            .selectFrom(stage)
            .orderBy(stage.name.asc())
            .fetch()
    }
}
